ow_faults/src/main.cpp

```diff
@@ -11,12 +12,14 @@ using namespace std;
 
 int main(int argc, char **argv) {
   ros::init(argc, argv, "fault");
-  ros::NodeHandle node_handle;
+  ros::NodeHandle node_handle_faults;
+  ros::NodeHandle node_handle_detector;
 
   dynamic_reconfigure::Server<ow_faults::FaultsConfig> server;
   dynamic_reconfigure::Server<ow_faults::FaultsConfig>::CallbackType f;
 
-  FaultInjector fault_injector(node_handle);
+  FaultInjector fault_injector(node_handle_faults);
+  FaultDetector fault_detector(node_handle_detector);
 
   f = bind(&FaultInjector::faultsConfigCb, &fault_injector, placeholders::_1, placeholders::_2);
   server.setCallback(f);
```

Comment:
I think this corresponds with what I have mentioned earlier. The fault detection should be a separate node (fully separate).